app/javascript/src/components/editor/SearchObjects.svelte

```diff
@@ -0,0 +1,18 @@
+<script>
+  export let objects = []
+  export let key = "title"
+  export let placeholder = "Search..."
+  export let filteredObjects = objects
+
+  let query = ""
+
+  $: if (objects) search(query)
+
+  function search() {
+    if (query) filteredObjects = objects.filter(o => o[key].toLowerCase().indexOf(query.toLowerCase()) !== -1)
+    else filteredObjects = objects
+
+  }
```

Comment:
Was it really not possible to make this a single `$:` statement somehow?

Reply:
It may have been! This feels a little neater though